### PR TITLE
Use localdb cache (localforage) for caching ref genome data

### PR DESCRIPTION
**Pull request by alisman**
Even when we pull this from CDN it still often takes seconds to download.

### CODE REVIEW

File: src/shared/lib/StoreUtils.ts [hunk 1]
```diff
@@ -232,19 +233,38 @@ export async function fetchAllReferenceGenomeGenes(
...
-        return $.ajax({
-            url: getFrontendAssetUrl('reactapp/reference_genome_hg19.json'),
-            dataType: 'json',
-        });
-    }
-    if (genomeName) {
-        return await internalClient.getAllReferenceGenomeGenesUsingGET({
-            genomeName: genomeName,
-        });
+    const doCaching = window.location.hostname.includes('.cbioportal.org');
```
Comment (alisman) [inline src/shared/lib/StoreUtils.ts:236]: shrug emoji

File: src/config/serverConfigDefaults.ts [hunk 2]
```diff
@@ -175,6 +175,8 @@ const ServerConfigDefaults: Partial<IServerConfig> = {
     saml_logout_local: false,
     patient_view_use_legacy_timeline: false,
     enable_request_body_gzip_compression: false,
+
+    referenceGenomeVersion: 'hg19',
```
Comment (inodb) [inline src/config/serverConfigDefaults.ts:179]: seems kinda weird to have this property, since one portal can have multiple reference genomes

**Comment by inodb** [comment]
@alisman - I don't fully follow why we need to store this in localStorage? Wouldn't built in browser caching solve this?

**Comment by alisman** [comment]
@inodb i think netlify blows away the browser cache every time we deploy.    unless it's smart enough to actually diff files.  if i'm right there, then it basically means people load this data about once a day.  sometimes, in my experiemnts, even from cdn, it takes a long time to load.

**Comment by alisman** [comment]
@inodb https://www.netlify.com/blog/2017/02/23/better-living-through-caching/

I'm pretty sure the etag is specific to release, not to each inidivuual file

**Comment by alisman** [comment]
@inodb a little more research makes me think maybe they do actually change EACH file's etag ONLY when they detect a change to it.

**Pull request closed without merging**